cli-tests/tests/llvm.test.ts

```diff
@@ -0,0 +1,114 @@
+import {executeCommand} from "../src/helper";
+import { paths } from '../src/entities';
+
+describe("Set of --llvm-ir tests", () => {
+  const zksolcCommand = 'zksolc';
+  const solcCommand = 'solc';
+
+  //id1744
+  describe(`Run ${zksolcCommand} with --llvm-ir by default`, () => {
+   /* const args = [`${paths.pathToLlvmContractsFile}`, `--llvm-ir`];*/
+    const args = [`${paths.pathToBasicLlvmContract}`, `--llvm-ir`];
+    const invalidArgs = [`--llvm-ir`];
+    const result = executeCommand(zksolcCommand, args);
+    const invalidResult = executeCommand(zksolcCommand, invalidArgs);
+
+    it("Valid command exit code = 0", () => {
+      expect(result.exitCode).toBe(0);
+    });
+
+    it("--llvm-ir output is presented", () => {
+      expect(result.output).toMatch(/(Compiler run successful. No output requested. Use --asm and --bin flags.)/i);
+    });
+
+    it("run invalid: zksolc --llvm-ir", () => {
+      expect(invalidResult.output).toMatch(/(The input file is missing)/i);
+    });
+
+    it("Invalid command exit code = 1", () => {
+      expect(invalidResult.exitCode).toBe(1);
+    });
+
+    it("Invalid solc exit code == Invalid zksolc exit code", () => {
+      const solcResult = executeCommand(solcCommand, invalidArgs);
+      expect(solcResult.exitCode).toBe(invalidResult.exitCode);
+    });
+  });
+
+  //id1825
+  describe(`Run ${zksolcCommand} with 2 same flags --llvm-ir --llvm-ir`, () => {
+    const args = [`${paths.pathToBasicLlvmContract}`, `--llvm-ir`, `--llvm-ir`];
+    const result = executeCommand(zksolcCommand, args);
+
+    it("Valid command exit code = 1", () => {
+      expect(result.exitCode).toBe(1);
+    });
+
+    it("--llvm-ir error is presented", () => {
+      expect(result.output).toMatch(/(The argument '--llvm-ir' was provided more than once)/i);
+    });
+
+    it("solc exit code == zksolc exit code", () => {
+      const solcResult = executeCommand(solcCommand, args);
+      expect(solcResult.exitCode).toBe(result.exitCode);
+    });
+  });
+
+  //id1826
+  describe(`Run ${zksolcCommand} with --llvm-ir with wrong input format`, () => {
+    const args = [`${paths.pathToBasicYulContract}`, `--llvm-ir`];
+    const result = executeCommand(zksolcCommand, args);
+
+    it("Valid command exit code = 1", () => {
+      expect(result.exitCode).toBe(1);
+    });
+
+    it("--llvm-ir error is presented", () => {
+      expect(result.output).toMatch(/(error: expected top-level entity)/i);
```

Comment:
@pcheremu here there's a strange behaviour.
In test case we have next line to expect:

```
output: solc error: Error: Expected keyword "object".
 --> input.sol:3:1:
  |
3 | pragma solidity >=0.0;
  | ^^^^^^
```

But in fact I have next one:
```
 "Contract `/***/era-compiler-solidity/cli-tests/src/contracts/yul/contract.yul` compiling error: /Users/olehbairak/GitHub/CyberShop/era-compiler-solidity/cli-tests/src/contracts/yul/contract.yul:1:1: error: expected top-level entity
    object \"Test\" {
    ^
    /Users/olehbairak/GitHub/CyberShop/era-compiler-solidity/cli-tests/src/contracts/yul/contract.yul:1:1: error: expected top-level entity
    object \"Test\" {
    ^"
```
Can you please check? Maybe we need to update test case?

Reply:
Sure, I will look at this

Reply:
@olehbairak the error above is for Solidity.
Invalid LLVM IR will produce a different one.

Reply:
Ok, thank you. So in fact, the error implemented is ok. Cause the test are implemented for LLVM IR.
@pcheremu

Reply:
@olehbairak , yep this warning belongs to sol contracts, so we are going to use it instead of Yul here. I have add changes : const args = [`${paths.pathToBasicSolContract}`, `--llvm-ir`]; Also I have specified it in the test documentation to avoid this confusing.